Util/ViolationFormatter.php

```diff
@@ -13,19 +13,21 @@
 
 use FOS\RestBundle\Controller\Annotations\Param;
 use FOS\RestBundle\Controller\Annotations\QueryParam;
-use FOS\RestBundle\Validator\ViolationFormatter as NewViolationFormatter;
 use Symfony\Component\Validator\ConstraintViolationInterface;
 use Symfony\Component\Validator\ConstraintViolationListInterface;
 
 /**
- * @deprecated since 1.8, to be removed in 2.0. Use {@link \FOS\RestBundle\Validator\ViolationFormatter} instead.
+ * @deprecated since version 1.8 and will be removed in 2.0. Catch specialized exception classes instead of relying on specific exception messages.
  */
 class ViolationFormatter implements ViolationFormatterInterface
 {
-    public function __construct()
+    private static $deprecationTriggered = false;
+
+    public function __construct($triggerDeprecation = true)
     {
-        if (!$this instanceof NewViolationFormatter) {
-            @trigger_error(sprintf('%s is deprecated since version 1.8. Use FOS\RestBundle\Validator\ViolationFormatter instead.', __CLASS__), E_USER_DEPRECATED);
+        if ($triggerDeprecation && !static::$deprecationTriggered) {
+            @trigger_error(sprintf('The %s class is deprecated since version 1.8 and will be removed in 2.0. Catch specialized exception classes instead of relying on specific exception messages.', __CLASS__), E_USER_DEPRECATED);
```

Comment:
I think this should be trigerred in the param fetcher instead so that we can still use this class internally.